### PR TITLE
chore(app-setting): app settings for ts function

### PR DESCRIPTION
**Pull request by formulahendry**
ADO task: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_git/teamsfx-specs/pullrequest/6505698?path=/specs/app-setting-improvement.md&_a=files

- Add config.ts to include all the configs
- Add .env.teamsfx.local to include all the configs

This PR is part of `app settings improvement` feature. Will update more templates.

Previous PR for ts sso-tab: https://github.com/OfficeDev/TeamsFx/pull/6249

### CODE REVIEW

File: templates/function-base/ts/default/.env.teamsfx.local
Comment (swatDong) [inline templates/function-base/ts/default/.env.teamsfx.local:1]: Use the updated comment "# TeamsFx will overwrite the following variable values when running debug"

Comment (formulahendry) [reply templates/function-base/ts/default/.env.teamsfx.local]: Good catch. Thx!

**Review by swatDong — APPROVED** [review]

**Comment by adashen** [comment]
@formulahendry  Please hold this PR until 4.1.0 rc is ready. The change should not be included in the coming release. Besides we need figure out the add sso/add api client experience first before merging the code.

**Pull request merged**
